### DISCUSSION TITLE
New dag file takes minutes to be recognized

### DISCUSSION THREAD

**Discussion by jerryluicy** [apache/airflow#26850] [Q&A]
Hi there,

I am running airflow in docker using the instructions here: https://airflow.apache.org/docs/apache-airflow/stable/howto/docker-compose/index.html

When I add a new dag file to the `dags` folder, it is very slow (few minutes) for it to be shown using the GetDags API (from golang airflow client). But it seems to be not specific to the golang client, because if I open a shell in the web server container, and run `airflow list dags` immediately after creating the new dag file, it will show my new dag file, but with the `paused` status being `None`, and it will change to `True` (or `false`) on its own in a few minutes. And only then, I can see my newly created dag from the GetDags API.
```
default@a29681f500c5:/opt/airflow$ airflow dags list
dag_id                            | filepath                       | owner   | paused
==================================+================================+=========+=======
my_simple_dag_id4                 | a4.py                          | airflow | None  

```
the dag file I am using: https://github.com/apache/airflow/blob/main/airflow/example_dags/tutorial.py

Is there any way to reduce the wait?

hoping to get the wait to within a few seconds 

Thanks

**Comment by Taragolis** [comment]
If you expected problems that changes in your DAGs folder not appear immediately you could redefine this environment variables (or change airflow.cfg) by required values:

`AIRFLOW__SCHEDULER__DAG_DIR_LIST_INTERVAL`  - https://airflow.apache.org/docs/apache-airflow/stable/configurations-ref.html#dag-dir-list-interval , by default Airflow scan for new files only once per 5 minutes
`AIRFLOW__SCHEDULER__MIN_FILE_PROCESS_INTERVAL` - https://airflow.apache.org/docs/apache-airflow/stable/configurations-ref.html#min-file-process-interval
`AIRFLOW__CORE__MIN_SERIALIZED_DAG_FETCH_INTERVAL` - https://airflow.apache.org/docs/apache-airflow/stable/configurations-ref.html#min-serialized-dag-fetch-interval
`AIRFLOW__CORE__MIN_SERIALIZED_DAG_UPDATE_INTERVAL ` - https://airflow.apache.org/docs/apache-airflow/stable/configurations-ref.html#min-serialized-dag-update-interval

In most cases this values fine for production, however in local development you might want to set lower values

**Comment by potiuk** [comment]
Also see this detailed discussion: https://airflow.apache.org/docs/apache-airflow/stable/concepts/scheduler.html?highlight=fine+tuning#fine-tuning-your-scheduler-performance
